hubapi/README.MD

```diff
@@ -0,0 +1,75 @@
+`HubAPI`
+The HubAPI supports operations for listing jina hub images
+and pushing images to the hub. These are authorised using the
+hub authorizer. We use MongoDB for storing and retrieving
+information related to hub. This document elaborates on the functioning
+of each of these APIs
+1. `Hub List` API
+List API provides the user the ability to query jina images. Under the hood, the implementation uses a Mongodb Handler to connect to the database & insert documents in the collection.
+Documents are generally manifest files that have metadata information of hub images. Currently, the List API is optimised for listing the images with the latest jina version.
+
+* API gateway endpoint: `/GET` `images`        
+* Supports cursor based pagination and alphabetical sorting
+* Returns only latest images for all executors (only one result per executor)
+* Returns docker pull command in addition to all info in manifest file.
+* Semver sort implementation with different fields for `major`, `minor` & `patch`
+* Optimized filter for searching with params:
+`kind` (indexer, encoder, segmenter etc),
+`type` (pod, app etc),
+`jina version`,
+`executor (manifest) version`,
+`keywords` (array search),
+`executor name` (simple regex search).
+  
+The `Aggregation` pipeline has the following stages:
+
```

Comment:
For this section, can we break stages down into their own sub-headings?

`Stage 1:` -> `### Stage 1`